tinystatus

```diff
@@ -87,42 +87,42 @@ li { list-style: none; margin-bottom: 2px; padding: 5px; border-bottom: 1px soli
 EOF
 outagenb="$(find "${tmp}/ko" -mindepth 1 | grep -c 'status$')"
 if [ "${outagenb}" -ne 0 ]; then
-    echo "<ul><li class='panel failed-bg'>${outagenb} Outage(s)</li></ul>"
+    echo "<ul><li class='panel failed-bg'>${outagenb} Outage(s)</li></ul>" >> "${tmp}/report/index.html"
 else
-    echo "<ul><li class='panel success-bg'>All Systems Operational</li></ul>"
+    echo "<ul><li class='panel success-bg'>All Systems Operational</li></ul>" >> "${tmp}/report/index.html"
 fi
-cat << EOF
+cat << EOF >> "${tmp}/report/index.html"
 <h1>Services</h1>
 <ul>
 EOF
 for file in "${tmp}/ko/"*.status; do
     [ -e "${file}" ] || continue
     name="$(basename "${file}" | sed 's,.status$,,')"
     status="$(cat "${file}")"
-    echo "<li>${name} <span class='small failed'>(${status})</span><span class='status failed'>Disrupted</span></li>"
+    echo "<li>${name} <span class='small failed'>(${status})</span><span class='status failed'>Disrupted</span></li>" >> "${tmp}/report/index.html"
 done
 for file in "${tmp}/ok/"*.status; do
     [ -e "${file}" ] || continue
     name="$(basename "${file}" | sed 's,.status$,,')"
-    echo "<li>${name} <span class='status success'>Operational</span></li>"
+    echo "<li>${name} <span class='status success'>Operational</span></li>" >> "${tmp}/report/index.html"
 done
-cat << EOF
+cat << EOF >> "${tmp}/report/index.html"
 </ul>
 <p class=small> Last check: $(date -I'seconds')</p>
 EOF
 if [ -f "${incidentsfile}" ]; then
-    echo '<h1>Incidents</h1>'
+    echo '<h1>Incidents</h1>' >> "${tmp}/report/index.html"
     if [ -s "${incidentsfile}" ]; then
-        sed 's|^\(.*\)$|<p>\1</p>|' "${incidentsfile}"
+        sed 's|^\(.*\)$|<p>\1</p>|' "${incidentsfile}" >> "${tmp}/report/index.html"
     else
-        echo '<p>No incident reported yet ;)</p>'
+        echo '<p>No incident reported yet ;)</p>' >> "${tmp}/report/index.html"
     fi
 fi
-cat <<EOF
+cat << EOF >> "${tmp}/report/index.html"
 </div>
 </body></html>
 EOF
-
+cat "${tmp}/report/index.html"
```

Comment:
Just out of curiosity, to see if I understand right: you came across a situation where the script aborted midway and output corrupt HTML, e.g. because of missing closing tags?

In that case, I don't know if this final `cat` will solve it. The script currently runs without `set -e`, so it happily continues on error regardless. Thus, intermediary steps might fail, but later steps still happily write to `"${tmp}/report/index.html"`, potentially producing the same corrupt output, only it is saved to a file first.

Reply:
> Just out of curiosity, to see if I understand right: you came across a situation where the script aborted midway and output corrupt HTML, e.g. because of missing closing tags?
No, but there was a situation where I wanted to use stdout for other purposes and have the file be copied to a destination by the script rather than using stdout. This approach seems cleaner in cases where unexpected output is produced. All of this is speculative but I figured I'd put it out there.